### PR TITLE
Add ability for a user to delete themselves

### PR DESCRIPTION
**Pull request by lostcodingsomewhere**
- Add service and controller for deleting a user
- This will mostly be useful for now for postman testing purposes

### CODE REVIEW

File: src/framework/controllers/WritableController.kt [hunk 1]
```diff
@@ -35,7 +35,7 @@ interface WritableController<T, U> {
      * @param user [U] User who is requesting (to verify permissions)
      * @param id Unique ID of the [T]
      */
-    fun delete(user: U, requestData: RequestData, id: Int): SOAResult<T> {
+    fun delete(user: U, requestData: RequestData): SOAResult<Boolean> {
```
Comment (nucat176) [inline src/framework/controllers/WritableController.kt]: Is boolean the normal result type for a deletion call in REST?

File: src/test/unit/services/user_account/DeleteUserAccountServiceTest.kt [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        Handler.connectAndBuildTables()
+    }
+
+    override fun afterTest(description: Description, result: TestResult) {
+        Handler.disconnectAndDropTables()
+    }
+
+    init {
+        "calling execute with a valid user account" should {
+            "return a success result and a deleted user account" {
```
Comment (nucat176) [inline src/test/unit/services/user_account/DeleteUserAccountServiceTest.kt]: it doesn't return the deleted user account?

**Comment by nucat176** [comment]
also theres a lot of onDelete=CASCADE additions in models/schema that arent related to the rest of this change. whats the rationale behind that?

File: src/framework/controllers/WritableController.kt (hunk 1 above)
Comment (lostcodingsomewhere) [reply src/framework/controllers/WritableController.kt]: usually an empty body, but i don't have that implemented for SoAResult type yet

File: src/test/unit/services/user_account/DeleteUserAccountServiceTest.kt (hunk 2 above)
Comment (lostcodingsomewhere) [reply src/test/unit/services/user_account/DeleteUserAccountServiceTest.kt]: ill fix that

**Comment by lostcodingsomewhere** [comment]
just need those once we start deleting, so i added them

**Pull request merged**